src/run.rs

```diff
@@ -88,17 +97,47 @@ pub fn image_ids() -> io::Result<HashSet<String>> {
             "Unable to determine IDs of all images.",
         ));
     }
-
+    let mut images = HashMap::new();
     // Interpret the output bytes as UTF-8 and collect the lines.
     String::from_utf8(output.stdout)
-        .map(|output| {
-            output
-                .lines()
-                .map(|line| line.trim().to_owned())
-                .filter(|line| !line.is_empty())
-                .collect()
-        })
         .map_err(|error| io::Error::new(io::ErrorKind::Other, error))
+        .and_then(|output| {
+            let lines = output.trim().lines();
+            for line in lines {
+                if line.is_empty() {
+                    continue;
+                }
+                let tab_index = line.find('\t').ok_or_else(|| {
+                    io::Error::new(io::ErrorKind::Other, "Failed to split image ID and date")
```

Comment:
For consistency with the existing error messages, can we add a period at the end of the string?